script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -102,60 +119,55 @@ class PublishPluginCommand extends PluginCommand {
     }
     _print('Local repo is ready!');
 
-    await _publish();
+    final Directory packageDir = _checkPackageDir(package);
+    await _publishPlugin(packageDir: packageDir);
+    await _tagRelease(packageDir: packageDir, remote: remote, remoteUrl: remoteUrl, shouldPushTag: shouldPushTag);
+    await _finishSuccesfully();
+  }
+
+  Future<void> _publishPlugin({@required Directory packageDir}) async {
+    await _checkGitStatus(packageDir);
+    await _publish(packageDir);
     _print('Package published!');
+  }
+
+  Future<void> _tagRelease({@required Directory packageDir, @required String remote, @required String remoteUrl, @required bool shouldPushTag}) async {
     if (!argResults[_tagReleaseOption]) {
```

Comment:
I think it would make more sense to wrap the call to _tagRelease in this check, rather than making the function silently no-op based on an argument.